src/components/DirectoryTable.tsx

```diff
@@ -0,0 +1,228 @@
+      const valueA = a[sortBy as keyof User]?.toString();
+      const valueB = b[sortBy as keyof User]?.toString();
+
+      // edge case that value is empty
+      if (!valueA) {
+        return 1;
+      }
+      if (!valueB) {
+        return -1;
+      }
+
+      if (payload.reversed) {
+        return valueB.localeCompare(valueA);
+      }
+
+      return valueA.localeCompare(valueB);
+    }),
+    payload.search,
+  );
+}
+
+export default function DirectoryTable({ data }: { data: User[] }) {
+  const [search, setSearch] = useState('');
+  const [sortedData, setSortedData] = useState<User[]>(data);
+  const [sortBy, setSortBy] = useState<keyof RowData | null>(null);
+  const [reverseSortDirection, setReverseSortDirection] = useState(false);
+
+  const setSorting = (field: keyof RowData) => {
+    const reversed = field === sortBy ? !reverseSortDirection : false;
+    setReverseSortDirection(reversed);
+    setSortBy(field);
+    setSortedData(sortData(data, { sortBy: field, reversed, search }));
+  };
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.currentTarget;
+    setSearch(value);
+    setSortedData(
+      sortData(data, { sortBy, reversed: reverseSortDirection, search: value }),
+    );
+  };
+
+  const rows = sortedData.map((row) => (
+    <Table.Tr key={row.firstName + '_' + row.lastName}>
+      <Table.Td>{row.firstName + ' ' + row.lastName}</Table.Td>
+      <Table.Td>
+        {row.role === 'admin' && (
+          <Pill size="md" className="bg-red-700 text-white">
+            Staff
+          </Pill>
+        )}
+        {row.role === 'mentor' && (
+          <Pill size="md" className=" bg-blue-700 text-white">
+            {capitalize(row.role)}
+          </Pill>
+        )}
+        {row.role === 'mentee' && (
+          <Pill size="md" className=" bg-sky-600 text-white">
+            {capitalize(row.role)}
+          </Pill>
+        )}
+      </Table.Td>
+      <Table.Td>{row.email.toLowerCase()}</Table.Td>
+      <Table.Td>{row.industry ? row.industry : 'N/A'}</Table.Td>
+      <Table.Td>{capitalize(row.gender)}</Table.Td>
+      <Table.Td>
+        <Button
+          size="sm"
+          color="blue"
+          variant="outline"
+          component={Link}
+          href={`/profile/${row._id}`}
+        >
+          View Profile
+        </Button>
+      </Table.Td>
+    </Table.Tr>
+  ));
+
+  return (
+    <ScrollArea>
+      <TextInput
+        size="lg"
+        placeholder="Search by any field"
+        mb="lg"
+        leftSection={<MagnifyingGlassIcon className="w-6 h-5" />}
+        value={search}
+        onChange={handleSearchChange}
+      />
+      <Table
+        horizontalSpacing="md"
+        verticalSpacing="xs"
+        miw={700}
+        layout="fixed"
+      >
+        <Table.Tbody>
+          <Table.Tr>
+            {tableHeaders.map((headerTitle) => {
+              return (
+                <Th
+                  key={headerTitle}
```

Comment:
what is headerTitle, the user' name? I think the keys should be the user ID if that is not the case - this might fix the bugged behavior mentioned before.

Reply:
Header title is column title. I fixed the naming bc I agree it is contrived. They key used for table rows was what caused the issue - changed it to user _id and it is not longer buggy. My b 🙏